OpenRA.Mods.Common/Traits/World/PathSearch.cs

```diff
@@ -143,145 +183,123 @@ public PathSearch FromPoint(CPos from)
 		// For horizontal/vertical directions, the set is the three cells 'ahead'. For diagonal directions, the set
 		// is the three cells ahead, plus the two cells to the side, which we cannot exclude without knowing if
 		// the cell directly between them and our parent is passable.
-		static CVec[][] directedNeighbors = {
-			new CVec[] { new CVec(-1, -1), new CVec(0, -1), new CVec(1, -1), new CVec(-1, 0), new CVec(-1, 1) },
-			new CVec[] { new CVec(-1, -1), new CVec(0, -1), new CVec(1, -1) },
-			new CVec[] { new CVec(-1, -1), new CVec(0, -1), new CVec(1, -1), new CVec(1, 0), new CVec(1, 1) },
-			new CVec[] { new CVec(-1, -1), new CVec(-1, 0), new CVec(-1, 1) },
+		static readonly CVec[][] DirectedNeighbors = {
+			new[] { new CVec(-1, -1), new CVec(0, -1), new CVec(1, -1), new CVec(-1, 0), new CVec(-1, 1) },
+			new[] { new CVec(-1, -1), new CVec(0, -1), new CVec(1, -1) },
+			new[] { new CVec(-1, -1), new CVec(0, -1), new CVec(1, -1), new CVec(1, 0), new CVec(1, 1) },
+			new[] { new CVec(-1, -1), new CVec(-1, 0), new CVec(-1, 1) },
 			CVec.Directions,
-			new CVec[] { new CVec(1, -1), new CVec(1, 0), new CVec(1, 1) },
-			new CVec[] { new CVec(-1, -1), new CVec(-1, 0), new CVec(-1, 1), new CVec(0, 1), new CVec(1, 1) },
-			new CVec[] { new CVec(-1, 1), new CVec(0, 1), new CVec(1, 1) },
-			new CVec[] { new CVec(1, -1), new CVec(1, 0), new CVec(-1, 1), new CVec(0, 1), new CVec(1, 1) },
+			new[] { new CVec(1, -1), new CVec(1, 0), new CVec(1, 1) },
+			new[] { new CVec(-1, -1), new CVec(-1, 0), new CVec(-1, 1), new CVec(0, 1), new CVec(1, 1) },
+			new[] { new CVec(-1, 1), new CVec(0, 1), new CVec(1, 1) },
+			new[] { new CVec(1, -1), new CVec(1, 0), new CVec(-1, 1), new CVec(0, 1), new CVec(1, 1) },
 		};
 
-		static CVec[] GetNeighbors(CPos p, CPos prev)
+		static IEnumerable<CVec> GetNeighbors(CPos p, CPos prev)
```

Comment:
Since this is private, it would be safe enough to keep the return type as an array. This avoids the performance penalty of enumerating via the IEnumarable interface.